pkg/promtail/targets/gcplog/metrics.go

```diff
@@ -0,0 +1,37 @@
+package gcplog
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// Metrics stores gcplog entry mertrics.
+type Metrics struct {
+	// reg is the Registerer used to create this set of metrics. May be nil.
+	reg prometheus.Registerer
+
+	gcplogEntries *prometheus.CounterVec
+	gcplogErrors  *prometheus.CounterVec
+}
+
+// NewMetrics creates a new set of metrics. If reg is non-nil, the metrics
+// will be registered.
+func NewMetrics(reg prometheus.Registerer) *Metrics {
+	var m Metrics
+	m.reg = reg
+
+	m.gcplogEntries = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "promtail",
+		Name:      "gcplog_target_entries_total",
+		Help:      "Help number of successful entries sent to the gcplog target",
+	}, []string{"project"})
+
+	m.gcplogErrors = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "promtail",
+		Name:      "gcplog_target_parsing_errors_total",
+		Help:      "Total number of parsing errors while receiving gcplog messages",
+	}, []string{"project"})
+
+	if reg != nil {
```

Comment:
Would there ever be a case where we would want to pass a nil registerer?
If not I think we should not support it and assume registerer would always be non nil to avoid bugs.
What do you think?

Reply:
I think I copied this pattern from Prometheus where it's nil just to make tests easier to write, but this makes sense, I'll change it.

Reply:
I have seen Cortex and Loki doing it like https://github.com/grafana/loki/blob/e59adccfa0bbdade7dfd7a1e0747c2629064e931/pkg/storage/stores/shipper/metrics.go#L14

Reply:
Gotcha. We can't do that exactly in promtail since we need to pull out the specific registerer used. I'm biased towards how I have it written it now because I don't like too much indentation, but I can change it to be similar to that if you want.